src/analysis/AnalysisNotificationManager.ts

```diff
@@ -120,30 +141,43 @@ const AppErrorNotification = ({ app }) => {
   ]);
 };
 
-function RuntimeManager({ namespace, name, runtimes, apps }) {
-  const prevRuntimes = usePrevious(runtimes);
-  const prevApps = usePrevious(apps);
+interface AnalysisNotificationManagerProps extends PropsWithChildren {
+  namespace: string;
+  name: string;
+  runtimes: Runtime[];
+  apps: App[];
+}
+
+export const AnalysisNotificationManager = (props: AnalysisNotificationManagerProps) => {
+  const { namespace, name, runtimes, apps } = props;
+  const prevRuntimes: Runtime[] = usePrevious(runtimes) || [];
+  const prevApps: App[] = usePrevious(apps) || [];
 
   useEffect(() => {
-    const runtime = getCurrentRuntime(runtimes) || {};
-    const prevRuntime = _.last(_.sortBy('auditInfo.createdDate', _.remove({ status: 'Deleting' }, prevRuntimes))) || {};
+    const runtime: Runtime | undefined = getCurrentRuntime(runtimes);
+    const prevRuntime: Runtime | undefined = _.last(
+      _.sortBy('auditInfo.createdDate', _.remove({ status: 'Deleting' }, prevRuntimes))
+    );
     const twoMonthsAgo = _.tap((d) => d.setMonth(d.getMonth() - 2), new Date());
     const welderCutOff = new Date('2019-08-01');
-    const createdDate = new Date(runtime?.createdDate);
+    const createdDate = runtime ? new Date(runtime.auditInfo.createdDate) : new Date();
```

Comment:
as always, typescript finds bugs. This code was not working before.... (notice the incorrect `runtime.createdDate` changed to `runtime.auditInfo.createdDate`)